section3/reflection.md

```diff
@@ -2,12 +2,24 @@
 
 1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
 
+-It certainly felt more challenging at times but I thought it was a great opportunity to learn how to research new concepts. Even though some of the examples in the documentation or others I could find online seemed a bit more complex than what I'm used to, it still helped a lot to read through. I learned several methods that could be applied to questions in the exercises that shortened the amount I needed to code if I had just stuck to the ones we were required to research! .splice and .map come to mind. I liked the more independent type of learning personally.
+
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
 
+-Spotify probably uses a lot of arrays! From albums themselves being arrays of songs, to playlists being arrays of songs, to artists as well being arrays. It is arrays all the way down! Even the ads that cycle are part of an array with maybe 4 different ads maximum being shuffled at any given time.
+
 3. In your own words, explain what an Array method is.
 
+-An Array method is a specific action that is called and performed on the Array. An Array method can insert elements into the Array, as well as remove or replace elements from any point. It can also sort, reverse, apply a function, as well as many more options that can be called to modify an Array.
+
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
 
+-A loop checks a condition and it will continue running as long as that condition remains true. They are usefull in conjunction with an array if you want to run the same code for each item in an array. If you don't know how many items will end up in an array when building a script you can set a counter to the total number of items in the array so it will run the correct amount of statements. It is important to know that if your loop contains a lot of items it can make the page slower to load. If a condition never returns false you can get what is known as an "infinite loop" which breaks the page. Anything that can be done outside of a loop should stay outside of a loop. It can be wasteful of resources to declare a variable needlessly within a loop if it doesn't have to be there.
+
 5. If you had to teach a 5 year old about loops, how would you explain it?
 
+-A loop is when you do the same thing over and over again. It's ilke the 99 bottles of pop on the wall song!
+
 6. What questions do you still have about Arrays and loops?
```

Comment:
Great question! The only difference is when the condition is checked. In a `while` loop, you check the condition first and then run the code. In a `do while` loop, you run the code first and then check the condition. This means that the code will always be executed at least once in a `do while` loop.